### PR TITLE
Types 2.0

### PR DESCRIPTION
**Pull request by ronzeidman**
case 2. Improvement to existing type definition.

Source: https://github.com/hapijs/joi/blob/master/API.md#errors
## Errors

Joi throws classical javascript `Error`s containing :
- `name` - `'ValidationError'`.
- `isJoi` - `true`.
- `details` - an array of errors :
  - `message` - string with a description of the error.
  - `path` - dotted path to the key where the error happened.
  - `type` - type of the error.
  - `context` - object providing context of the error.
- `annotate` - function that returns a string with an annotated version of the object pointing at the places where errors occured.
- `_object` - the original object to validate.

### CODE REVIEW

File: joi/index.d.ts [hunk 1]
```diff
@@ -121,17 +121,17 @@ export interface IPOptions {
...
-    annotated(): string;
+    annotate(): string;
+    _object: any;
 }
 
 export interface ValidationErrorItem {
     message: string;
     type: string;
     path: string;
-    options?: ValidationOptions;
```
Comment (mhegazy) [inline joi/index.d.ts]: looks like this does exist, see https://github.com/hapijs/joi/blob/a2b0b8d43b6b8c0b0b556ceeb1537c6a1d67be4f/lib/errors.js#L55

**Comment by mhegazy** [comment]
@Bartvds, @aurence-myers, @cglantschnig>, @DavidBR-SW, and @GaelMagnan can you please review.

File: joi/index.d.ts (hunk 1 above)
Comment (ronzeidman) [inline joi/index.d.ts]: You're right, updated

**Comment by yuit** [comment]
Thanks @ronzeidman for the PR.

**Pull request merged**